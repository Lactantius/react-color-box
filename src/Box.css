.Box-remove-button {
  float: right;
  color: white;
  background-color: rgba(0, 0, 0, 0.2);
  border: none;
}

.Box-remove-button:hover {
  background-color: rgba(0, 0, 0, 0.8);
}
