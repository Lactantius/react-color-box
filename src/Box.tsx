import React from "react";

interface BoxStyle {
  width: string;
  height: string;
  color: string;
  id?: string;
}

function Box({ width, height, color }: BoxStyle): JSX.Element {
  return (
    <div
      className="Box"
      style={{ width: width, height: height, backgroundColor: color }}
    ></div>
  );
}

export default Box;
export type { BoxStyle };
