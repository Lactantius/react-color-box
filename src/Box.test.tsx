import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

const boxDefaults = (
  <Box width="50px" height="50px" color="#424242" remove={() => null} />
);

it("renders", () => {
  render(boxDefaults);
});

it("matches snapshot", () => {
  const { asFragment } = render(boxDefaults);
  expect(asFragment()).toMatchSnapshot();
});
