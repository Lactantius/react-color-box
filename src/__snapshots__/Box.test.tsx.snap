// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="width: 50px; height: 50px; background-color: rgb(66, 66, 66);"
  >
    <button
      class="Box-remove-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
